src/signals-unix.c

```diff
@@ -700,12 +707,97 @@ void trigger_profile_peek(void)
         }
     }
     bt_size_cur = 0; // clear profile buffer
-    if (jl_profile_start_timer() < 0)
+    if (jl_profile_start_timer(0) < 0)
         jl_safe_printf("ERROR: Could not start profile timer\n");
     else
         profile_autostop_time = jl_hrtime() + (profile_peek_duration * 1e9);
 }
 
+void jl_profile_task_unix(size_t nthreads)
+{
+    if (jl_profile_is_buffer_full()) {
+        // Buffer full: Delete the timer
+        jl_profile_stop_timer();
+        return;
+    }
+
+    jl_task_t *t = jl_get_random_task();
+    assert(t == NULL || jl_is_task(t));
+    if (t == NULL) {
+        return;
+    }
+    int t_state = jl_atomic_load_relaxed(&t->_state);
+    if (t_state == JL_TASK_STATE_DONE) {
+        return;
+    }
+
+    jl_rec_backtrace(t);
+
+    // store threadid but add 1 as 0 is preserved to indicate end of block
+    bt_data_prof[bt_size_cur++].uintptr = 1; // dummy value for now... Is this ever used when outputting the profile?
+
+    // store task id (never null)
+    bt_data_prof[bt_size_cur++].jlvalue = (jl_value_t*)t;
+
+    // store cpu cycle clock. XXX(Diogo, Nick): why are we recording the cycleclock here?
+    bt_data_prof[bt_size_cur++].uintptr = cycleclock();
+
+    // store whether thread is sleeping but add 1 as 0 is preserved to indicate end of block
+    bt_data_prof[bt_size_cur++].uintptr = 1; // dummy value for now... Is this ever used when outputting the profile?
```

Comment:
i think this might be used by `Profile.print` to output `Utilization: X% across all threads and tasks.`

Reply:
From discussion: augment this so that thread profilers are restricted to using a "sleeping state" of 1 and 2, and task profilers need to necessarily use a "fake sleeping state" of 3.

Then whenever `Profile.print` sees the profiler buffer, it's going to use this value to decide whether it should actually output something like "Utilization: X% across all threads and tasks.".